frontend/packages/dev-console/src/utils/truncate-utils.ts

```diff
@@ -0,0 +1,43 @@
+const DEFAULT_OPTIONS = {
+  length: 20,
+  middle: true,
+  omission: '\u2026', // ellipsis character
+  minTruncateChars: 3,
+};
+
+// Truncates a string down to `maxLength` characters when the length
+// is greater than the `maxLength` + `minTruncateChars` values.
+// By default the middle is truncated, set the options.middle to false to truncate at the end.
+// Truncated text is replaced with the provided omission option (ellipsis character by default);
+export const truncateText = (text, options = {}): string => {
+  const { length, middle, omission, minTruncateChars } = Object.assign(
+    {},
+    DEFAULT_OPTIONS,
+    options,
+  );
+
+  // Do not truncate less than the minimum truncate characters
+  if (text.length <= length + minTruncateChars) {
+    return text;
+  }
+
+  if (length <= omission.length) {
+    return omission;
+  }
+
+  if (middle) {
+    const startLength = Math.ceil((length - omission.length) / 2);
+    const endLength = length - startLength - omission.length;
+    const startFragment = text.substr(0, startLength);
+    const endFragment = text.substr(text.length - endLength);
+    return `${startFragment}${omission}${endFragment}`;
+  }
+
+  return `${text.substr(0, length - 1)}${omission}`;
+};
+
+export const shouldTruncate = (text, options = {}): boolean => {
+  const { length, minTruncateChars } = Object.assign({}, DEFAULT_OPTIONS, options);
```

Comment:
nit: I prefer the es6 spread syntax:
```suggestion
  const { length, minTruncateChars } = {...DEFAULT_OPTIONS, ...options};
```